crates/astria-sequencer/src/app/mod.rs

```diff
@@ -1333,3 +1195,243 @@ struct PostTransactionExecutionResult {
     validator_updates: Vec<tendermint::validator::Update>,
     consensus_param_updates: Option<tendermint::consensus::Params>,
 }
+
+#[derive(PartialEq)]
+enum ExitContinue {
+    Exit,
+    Continue,
+}
+
+enum Proposal<'a> {
+    Prepare(PrepareProposalInformation<'a>),
+    Process(ProcessProposalInformation),
+}
```

Comment:
I'm not sure there's much benefit to making separate structs for each of the variants.  I guess I'd have just flattened all this into the enum:

```suggestion
enum Proposal {
    Prepare {
        validated_txs: Vec<bytes::Bytes>,
        included_signed_txs: Vec<Transaction>,
        failed_tx_count: usize,
        execution_results: Vec<ExecTxResult>,
        excluded_txs: usize,
        current_tx_group: Group,
        mempool: Mempool,
        metrics: &'static Metrics,
    },
    Process {
        execution_results: Vec<ExecTxResult>,
        current_tx_group: Group,
    },
}
```